src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs

```diff
@@ -30,12 +30,29 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
         .RequireAuthorization()
         .WithParameterValidation();
 
-    endpointRouteBuilder.MapPost("/api/applications", async (ApplicationSubmissionRequest request, IMessageBus messageBus) =>
+    endpointRouteBuilder.MapPost("/api/applications", async Task<Results<Ok<SubmitApplicationResponse>, BadRequest<string>>> (ApplicationSubmissionRequest request, HttpContext ctx, IMessageBus messageBus) =>
         {
-          var cmd = new SubmitApplicationCommand(request.Id);
-          await messageBus.InvokeAsync<string>(cmd);
+          var userId = ctx.User.GetUserContext()?.UserId;
+          bool IdIsNotGuid = !Guid.TryParse(request.Id, out _); if (IdIsNotGuid) { return TypedResults.BadRequest("ApplicationId is not valid"); }
+          var query = new ApplicationsQuery
+          {
+            ById = request.Id,
+            ByApplicantId = userId,
+            ByStatus = new List<Managers.Registry.Contract.Applications.ApplicationStatus>() { Managers.Registry.Contract.Applications.ApplicationStatus.Draft }
+          };
+          var results = await messageBus.InvokeAsync<ApplicationsQueryResults>(query);
+          if (!results.Items.Any())
+          {
+            return TypedResults.BadRequest("Application not found");
+          }
 
-          return TypedResults.Ok();
+          var cmd = new SubmitApplicationCommand(request.Id);
+          var result = await messageBus.InvokeAsync<ApplicationSubmissionResult>(cmd);
+          if (!result.IsSuccess)
+          {
+            return TypedResults.BadRequest($"Application submission failed: {string.Join(',', result.ValidationErrors!)}");
```

Comment:
Worth looking how it shows up in axios - might need to return a problemdetail instance